### DISCUSSION TITLE
How to get queue info? (list of songs currently in queue)

### DISCUSSION THREAD

**Discussion by QuditWolf** [pear-devs/pear-desktop#1143] [General]
How do you handle queues? 

I don't know much of js and I can't figure out how queues work,

on the youtube videos website and music website there is a queue link...(however I don't think the link contains all the songs)
for example: https://music.youtube.com/watch?v=yvqHzdk8ZLc&list=OLAK5uy_lOxke5jTh9mQExan6XnCLXVoy69lHN11c

someway I can get an array of the songs in my queue?
then we could use the api to add them all to a playlist.

I'm trying to figure out how one would go about [this feature request](https://github.com/th-ch/youtube-music/issues/1130#issue-1693359694)

basically this feature:

![image](https://github.com/th-ch/youtube-music/assets/81188667/70f175ce-e2d6-469d-9f95-0a5178815d51)

this saves existing queue to a playlist on android(and ios too i think).
then we could save our carefully curated queues which are lost otherwise.

thx

**Comment by JellyBrick** [comment]
see https://github.com/th-ch/youtube-music/pull/1562

Getting the current queue in YTM is quite complicated. Hopefully, this code will help you.
